ros/api/v1/hosts.py

```diff
@@ -466,6 +501,31 @@ def get(self):
         io_states_dict['undersized'] = -1
         total_conditions = totals['cpu'] + totals['memory'] + totals['io']
 
+        current_utc_datetime = datetime.now(timezone.utc)
+        upload_date = (current_utc_datetime - timedelta(days=1)).date()
+
+        current_performance_profiles = [
+            record
+            for record in systems_with_performance_record_queryset
+            if record.report_date.date() == upload_date
```

Comment:
Sorry to asking again but I don't completely remember, Are we doing `date - 1` only to avoid the systems with state "Waiting for data"? Or Is there any other reason too?

Reply:
Np, we had run into a problem to select the date for the latest upload.

I was thinking this date could be dynamic but then there was no way to actually ascertain the correct latest upload date.

Hence, it was decided that we would use a static date i.e. yesterday's date.

Reply:
There is one more thing we could do is request a date from the client up until which he/she wants to see the report.

But, I think we will have to take that up in the next release.

Reply:
That said showing the report based on performance profiles only for one day feels a bit wrong, let me check with Pau

Reply:
@patilsuraj767 Change pushed, current profiles selection between today and stale_date